pkg/restclient/common.go

```diff
@@ -22,24 +22,18 @@ const (
 // ClientFunc receives a Client and overrides its members
 type ClientFunc func(*Client) error
 
-// WithTLSConfig overrides http.Client member with a client created
-// using specified TLS configuration.
+// WithTLSConfig applies tls config to underlying http.Client Transport
 func WithTLSConfig(tlsOpts *TLSOptions) ClientFunc {
 	return func(client *Client) error {
 		tlsConfig, err := NewTLSConfig(tlsOpts)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to create tlsconfig, err: %s", err.Error())
 		}
-		tr := &http.Transport{
-			DisableCompression: true,
-			DisableKeepAlives:  true,
```

Comment:
@aravindavk @prashanthpai  why we are disabling keep alives?  it will prevent user from re-using of TCP connections
between different HTTP requests. But we can reuse the TCP connection.

Reply:
There's no good reason to do it in restclient level. I'm assuming it was there because CLI was the only consumer of restclient and that doesn't need to have keep alive connections. The CLI command usually does one HTTP request and exits.